src/my_os2cond.c

```diff
@@ -166,9 +187,17 @@ int pthread_cond_timedwait(pthread_cond_t *cond, pthread_mutex_t *mutex,
 
 	cv = *cond;
 
-	_ftime(&curtime);
-	timeout= ((long) (abstime->tv_sec - curtime.time)*1000L +
+	if (!cv->attr->clock)
+	{
+		_ftime(&curtime);
+		timeout= ((long) (abstime->tv_sec - curtime.time)*1000L +
 			(long)((abstime->tv_nsec/1000) - curtime.millitm)/1000L);
+	}else{
+		clock_gettime (CLOCK_MONOTONIC, &ts);
+		timeout= ((long) (abstime->tv_sec - ts.tv_sec)*1000L +
+			(long)((abstime->tv_nsec - ts.tv_nsec)/1000)/1000L);
+	}
+

```

Comment:
and this all will become just a single block with a call to `clock_gettime (cv->attr->clock_id)...` (like in `else` clause).